src/v/cluster/scheduling/leader_balancer_constraints.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "cluster/scheduling/leader_balancer_constraints.h"
+
+namespace cluster::leader_balancer_types {
+
+std::optional<reassignment>
+even_shard_load_constraint::recommended_reassignment() {
```

Comment:
If this exists only for backporting purposes in older branches, wondering if we should omit it in `dev`. It seems like the new approach is superior and this code can bit rot.

Reply:
Agreed, the only caveat I can think of that since we are adding the `leader_balancer_mode` config option what happens when a used selects the older balancing method then upgrades? Should we just ignore the `leader_balancer_mode` option in newer RP versions?

Reply:
I'd consider this an "expert configuration", highly doubt if any user want to touch it. So I feel its okay to ignore after upgrade. I'll let other reviewers chime in too.